api/k0smotron.io/v1beta1/k0smotroncluster_types.go

```diff
@@ -196,6 +196,10 @@ type PersistenceSpec struct {
 	// PersistentVolumeClaim defines the PVC configuration. Will be used as is in case of .spec.persistence.type is pvc.
 	//+kubebuilder:validation:Optional
 	PersistentVolumeClaim *PersistentVolumeClaim `json:"persistentVolumeClaim,omitempty"`
+	// PVCDeletionAllowed defines whether the PVC should be deleted when the cluster is deleted.
+	//+kubebuilder:default=false
+	//+kubebuilder:validation:Optional
+	PVCDeletionAllowed bool `json:"pvcDeletionAllowed,omitempty"`
```

Comment:
IMO the name does not fully match the behaviour. Some alternative suggestions:
- `DeleteWithCluster`
- `DeletePVCsWithCluster`
- `AutoDeletePVCs`